altair/api.py

```diff
@@ -409,6 +409,70 @@ def to_dict(self):
         D['config'] = self.vlconfig.to_dict()
         return D
 
+    def select_x(self):
+        """
+        Helper function that does a best effort of selecting an automatic x axis.
+
+        Raises AssertionError if it cannot find a x axis to choose.
+        """
+        assert self.data is not None
+        assert len(self.data) >= 1
+
+        d = self._classify_data_by_type()
+
+        # Choose the first column found on the following order: T, O, N, Q
+        chosen_x = None
+        for typ in ['Q', 'N', 'O', 'T']:
+            if len(d[typ]) >= 1:
+                chosen_x = d[typ][0]
+
+        if chosen_x is None:
+            raise AssertionError('Could not select X.')
```

Comment:
Thanks for reviewing @jakevdp!

I had the break statement first but I'm generally against them. I would've used it if the list could be arbitrarily large, but it isn't here :) I also considered writing the list in order and calling reverse() on it but thought it unnecessary. Now I'm wondering if it is unnecessary or not :P

I used AssertionError because I like the assert syntax to validate assumptions. I can leave as is, change all exceptions thrown to be ValueError or do a mixture (AssertionError for assumptions up top and ValueError for the for loop. I like it as it is, but what do you think?